include/ck/utility/dynamic_buffer.hpp

```diff
@@ -321,7 +321,18 @@ struct DynamicBuffer
         {
             if(is_valid_element)
             {
-                atomic_add<X>(c_style_pointer_cast<X*>(&p_data_[i]), x);
+                constexpr index_t t_per_x = scalar_per_x_vector / scalar_per_t_vector;
+
+                if constexpr(t_per_x > 1)
+                {
+                    static_for<0, t_per_x, 1>{}([&](auto K) constexpr {
+                        atomicAdd(c_style_pointer_cast<scalar_t*>(&p_data_[i + K]), x[K()]);
```

Comment:
Please use atomic_add in dynamic_buffer here, instead of use hip defined atomicAdd.

It's on purpuse to not use hip defined atomicAdd function here, because there are overload functions. Using it may cause unexpected type casting

```atomic_add``` is used to avoid this unexpected type casting

Reply:
I see.    When the codes was written,  `include/ck/utility/generic_memory_space_atomic_add.hpp` is not available yet.

Reply:
Fixed by [3e0d5](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/237/commits/41673f814d77ab0889f2386383d0c902e5e3e0d5)